dynamic/dynamic_message.go

```diff
@@ -1983,67 +1984,85 @@ func validElementFieldValueForRv(fd *desc.FieldDescriptor, val reflect.Value) (i
 	}
 }
 
-func toInt32(v reflect.Value, what string, fieldName string) (int32, error) {
+func toInt32(v reflect.Value, fd *desc.FieldDescriptor) (int32, error) {
 	if v.Kind() == reflect.Int32 {
 		return int32(v.Int()), nil
 	}
-	return 0, fmt.Errorf("%s field %s is not compatible with value of type %v", what, fieldName, v.Type())
+	return 0, fmt.Errorf(
+		"%s field %s is not compatible with value of type %v",
```

Comment:
I'm now seeing that there was a lot of copy-pasta in this code before (my bad... we can all be lazy programmers sometimes).

But, since you're touching every one of these lines in this PR, let's take a second to consolidate. It looks like every error message here and below is the exact same, even same format arguments. So let's add a new function for this:

```go
func typeError(fd *desc.FieldDescriptor, t reflect.Type) error {
    return fmt.Errorf("%s field %s is not compatible with value of type %v",
        getTypeString(fd), fd.GetFullyQualifiedName(), t)
}
```

Then all of these duplicate lines become simpler/shorter:

```go
return <zero-value>, typeError(fd, v.Type())
```